### PR TITLE
Added support for AutoDiffScalar inputs to DrakeJoint methods

### PR DESCRIPTION
**Pull request by tkoolen**
Part of #1199, using CRTP.
@psiorx did most of the work.

### CODE REVIEW

**Comment by tkoolen** [comment]
This branch depends on the new Eigen version (https://github.com/RobotLocomotion/drake/issues/1213, see https://github.com/RobotLocomotion/eigen-pod/pull/1). Eigen has probably not been updated on drake006. @RussTedrake, could you pull the latest version of eigen?

**Comment by tkoolen** [comment]
Thanks!

**Comment by tkoolen** [comment]
@RussTedrake, I don't think that worked. I ssh'd into the machine and inspected /tmp/drake-master/eigen/eigen-eigen-10219c95fe65/Eigen/src/Core/util/Macros.h, which contains macros for the Eigen version number digits. EIGEN_MINOR_VERSION is set to 4 there, while on my machine it reads 5. I can see that the Makefile was updated so that it downloads 3.2.5, but maybe eigen never got built after that?

**Comment by tkoolen** [comment]
Looks good now, thanks.

**Comment by tkoolen** [comment]
runMaxROA doesn't pass for the same reason as in https://github.com/RobotLocomotion/drake/pull/1190. See issue https://github.com/RobotLocomotion/drake/issues/1197.

File: drake/systems/plants/joints/RevoluteJoint.h [hunk 1]
```diff
@@ -2,8 +2,9 @@
 #define REVOLUTEJOINT_H_
 
 #include "FixedAxisOneDoFJoint.h"
+#include <Eigen/Geometry>
 
-class DLLEXPORT_DRAKEJOINT RevoluteJoint: public FixedAxisOneDoFJoint
+class DLLEXPORT_DRAKEJOINT RevoluteJoint: public FixedAxisOneDoFJoint<RevoluteJoint>
```
Comment (RussTedrake) [inline drake/systems/plants/joints/RevoluteJoint.h]: regarding #1273 , i see that this second-level derived class doesn't overload any of the methods from FixedAxisOneDoFJoint.  I still think that you would have trouble doing that with this pattern.  E.g., what if you wanted to overload v2qdot here?  the compiler would complain that you're shadowing a non-virtual method.  no?

**Pull request merged**